pyan/sphinx.py

```diff
@@ -0,0 +1,169 @@
+"""
+Simple sphinx extension that allows including callgraphs in documentation.
+
+Example usage:
+
+```
+.. callgraph:: <function_name>
+
+
+Options are
+
+- **:no-groups:** (boolean flag): do not group
+- **:no-defines:** (boolean flag): if to not draw edges that show which
+  functions, methods and classes are defined by a class or module
+- **:no-uses:** (boolean flag): if to not draw edges that show how a function
+  uses other functions
+- **:no-colors:** (boolean flag): if to not color in callgraph (default is
+  coloring)
+- **:nested-grops:** (boolean flag): if to group by modules and submodules
+- **:annotated:** (boolean flag): annotate callgraph with file names
+- **:direction:** (string): "horizontal" or "vertical" callgraph
+- **:toctree:** (string): path to toctree (as used with autosummary) to link
+  elements of callgraph to documentation (makes all nodes clickable)
+- **:zoomable:** (boolean flag): enables users to zoom and pan callgraph
+```
+"""
+import re
+from typing import Any
+
+from docutils.parsers.rst import directives
+from pyan import create_callgraph
+from sphinx.ext.graphviz import align_spec, figure_wrapper, graphviz
+from sphinx.util.docutils import SphinxDirective
+
+
+def direction_spec(argument: Any) -> str:
```

Comment:
I'm pretty much in the dynamically typed camp myself, so please correct if I'm wrong - I suppose the point of the type specification here is to make the return type explicit, since the input is left as `Any`?

Reply:
I believe so. Effectively, I pretty much copied and adopted that one-liner function from the official sphinx repo. So guess it is fine.

Reply:
Yeah, it's fine. Thanks for the sphinx extension!